### PR TITLE
Avoid CFAllocator lookup overhead in String bridging

### PR DESCRIPTION
**Pull request by Catfish-Man**
Fixes rdar://problem/49925943

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

**Comment by Catfish-Man** [comment]
Yah. I should build it locally before running benchmarks apparently though :D

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

**Comment by Catfish-Man** [comment]
All the regressed tests have nothing to do with the code that changed, so I think this is a win

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test

File: stdlib/public/stubs/FoundationHelpers.mm [hunk 1]
```diff
@@ -110,15 +112,15 @@ static CFRange cast(_swift_shims_CFRange value) {
 _swift_shims_CFStringRef
 swift::_swift_stdlib_CFStringCreateCopy(_swift_shims_CFAllocatorRef alloc,
                                         _swift_shims_CFStringRef theString) {
-  return cast(CFStringCreateCopy(cast(alloc), cast(theString)));
+  return cast(CFStringCreateCopy(kCFAllocatorSystemDefault, cast(theString)));
```
Comment (milseman) [inline stdlib/public/stubs/FoundationHelpers.mm]: Can you assert the alloc is null?

File: stdlib/public/stubs/FoundationHelpers.mm [hunk 2]
```diff
@@ -110,15 +112,15 @@ static CFRange cast(_swift_shims_CFRange value) {
...
 }
 
 _swift_shims_CFStringRef
 swift::_swift_stdlib_CFStringCreateWithBytes(
     _swift_shims_CFAllocatorRef _Nullable alloc, const uint8_t *bytes,
     _swift_shims_CFIndex numBytes, _swift_shims_CFStringEncoding encoding,
     _swift_shims_Boolean isExternalRepresentation) {
-  return cast(CFStringCreateWithBytes(cast(alloc), bytes, numBytes,
+  return cast(CFStringCreateWithBytes(kCFAllocatorSystemDefault, bytes, numBytes,
```
Comment (milseman) [inline stdlib/public/stubs/FoundationHelpers.mm]: Same

**Review by milseman — APPROVED** [review]
Please add an assertion in case someone tries to call it with a real allocator and intended it. Otherwise, LGTM

File: stdlib/public/stubs/FoundationHelpers.mm (hunk 2 above)
Comment (jckarter) [reply stdlib/public/stubs/FoundationHelpers.mm]: Can we remove the CFAllocatorRef argument entirely from the shim?

Comment (milseman) [reply stdlib/public/stubs/FoundationHelpers.mm]: (I was assuming shims were ABI, but I guess not?)

Comment (Catfish-Man) [reply stdlib/public/stubs/FoundationHelpers.mm]: I'm pretty sure they're ABI.

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test and merge

File: stdlib/public/stubs/FoundationHelpers.mm (hunk 2 above)
Comment (jckarter) [reply stdlib/public/stubs/FoundationHelpers.mm]: That'd be extremely unfortunate if they are. But if so, can we still replace the allocator argument with a `const void *unused` if we know we never use it?

Comment (Catfish-Man) [reply stdlib/public/stubs/FoundationHelpers.mm]: Sure

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test and merge

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test and merge

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test and merge

**Pull request merged**